images/router/haproxy/conf/haproxy-config.template

```diff
@@ -439,6 +442,15 @@ backend {{genBackendNamePrefix $cfg.TLSTermination}}:{{$cfgIdx}}
   http-request add-header Forwarded for=%[src];host=%[req.hdr(host)];proto=%[req.hdr(X-Forwarded-Proto)]
   {{- end }}
 
+  {{- with $pathRewriteTarget := firstMatch $pathRewriteTargetPattern (index $cfg.Annotations "haproxy.router.openshift.io/rewrite-target") }}
+  # Path rewrite target
+    {{- if eq $pathRewriteTarget "/" }}
```

Comment:
Why is this if required?  

If we could leave it to a simple pattern without side effect, I believe it should help in the long term.  

I may not be understanding why "/" is a special case though.

(Thanks for looking into this :) ).

Reply:
Is it that we're trying to allow /$1 and / to do the same thing?

Reply:
Hello,
it's to avoid the missing `/` or the double `/` in the final request to the backend server.

Reply:
https://haproxy-ingress.github.io/docs/configuration/keys/#rewrite-target doesn't seems to consider this a special case.

Reply:
I think I just understood what it meant... Line 2 and line 5 in the haproxy example seems to consider it a special case to be handle only when you get "//" or "" so that the output is "/".   Those special case would be handled by the defautl behaviour of haproxy (point 1)

In here, we only handle the case where the rewrite path annotation would be "/" so that all traffic to the host only (i.e. http://endpoint) would still be routed correctly to the backend service (i.e. http://endpoint/).  is that it? (Point2)

Reply:
(I'm still wondering about Point2 being right or not for my own understanding XD.  Awaiting your answer anxiously :) )